Source/Api/Analytics.swift

```diff
@@ -46,21 +47,50 @@ public class Analytics {
 	
 	/**
 		Need to call method to initialize the library
-		
+
 		- Throws: `AnalyticsError.analyticsAlreadyInitialized` if the Analytics
 		library is already initialized.
+		- Throws: `AnalyticsError.dataSourceIdNullOrEmpty` if com.liferay.analytics.DataSourceId
+		wasn't filled.
+		- Throws: `AnalyticsError.invalidEndpointURL` if com.liferay.analytics.DataSourceId
+		wasn't filled or value don't is a valid URL.
+		- Throws: `AnalyticsError.invalidFlushIntervalValue` if flushInterval be less then 1.
 	*/
-	public class func configure(
-		analyticsKey: String, flushInterval: Int = Analytics.DEFAUL_TIME_INTERVAL) throws {
-		
+	public class func `init`(flushInterval: Int = Analytics.DEFAUL_TIME_INTERVAL) throws {
+        
 		if let _ = sharedInstance {
 			throw AnalyticsError.analyticsAlreadyInitialized
 		}
-		
+        
 		let fileStorage = try FileStorage()
-		
-		sharedInstance = Analytics(
-			analyticsKey: analyticsKey, flushInterval: flushInterval, fileStorage: fileStorage)
+
+		let bundle = Bundle(for: self)
+		var settings: [String: String]?
+
+		if let path = bundle.path(forResource: "Info", ofType:"plist") {
+			settings = NSDictionary(contentsOfFile: path) as? [String: String]
+		}
+
+		guard let dataSourceId = settings?["com.liferay.analytics.DataSourceId"] else {
+			throw AnalyticsError.dataSourceIdNullOrEmpty
+		}
+
+		var endpointURL: String = ""
+
+		if let stringURL = settings?["com.liferay.analytics.EndpointUrl"] {
+			guard let _ = URL(string: stringURL) else {
```

Comment:
Would be better to unify the if let with the guard let.

Reply:
Yeah Paulo. I didn't knew that can be done. Thanks for warning and help me. I already the guard let.